docs/source/daml/intro/5_Restrictions.rst

```diff
@@ -0,0 +1,60 @@
+.. Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Restricting the Contract Model
+==============================
+
+Oftentimes, contract models will have constraints on the data stored or the allowed data transformations. In this section, you will learn about the two main mechanisms provided in DAML:
+
+1. The ``ensure`` keyword.
+2. The ``assert``, ``abort`` and ``error`` keywords.
+
+To make sense of the latter, you'll also learn more about the ``Update`` type and ``do`` blocks, which will be good preparation for :doc:`7_Composing`, where you will use ``do`` blocks to compose choices into complex transactions.
+
+Lastly, you will learn about time on the ledger and in scenarios.
+
+Template Pre-Conditions
+-----------------------
+
+The first kind of restriction one commonly puts on the contract model are called *template pre-conditions*. These are simply restrictions on the data that can be stored on a contract from that template. Suppose, for examples, that the ``SimpleIou`` contract from :ref:`simple_iou` should only be able to store positive amounts. This can be enforced using the ``ensure`` keyword.
```

Comment:
```suggestion
The first kind of restriction one commonly puts on contracts are called *template pre-conditions*. These are simply restrictions on the data that can be stored on a contract from that template. Suppose, for examples, that the ``SimpleIou`` contract from :ref:`simple_iou` should only be able to store positive amounts. This can be enforced using the ``ensure`` keyword.
```

It may be just me but "the contract model", singular, evokes something more fundamental, such as the abstract class of which each template is an instance.